### PR TITLE
Siddhi-execution-graph

### PR DESCRIPTION
**Pull request by NadeekaHP**
## Automation tests
 - Unit tests 
   > 92%

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        </dependencies>
+    </dependencyManagement>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-release-plugin</artifactId>
+                <configuration>
+                    <preparationGoals>clean install</preparationGoals>
```
Comment (Anoukh) [inline pom.xml]: Should run the documentation-deploy profile in the component pom

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    <autoVersionSubmodules>true</autoVersionSubmodules>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.wso2.siddhi</groupId>
+                <artifactId>siddhi-doc-gen</artifactId>
```
Comment (Anoukh) [inline pom.xml]: This plugins should be in the component pom

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <modules>
+                <module>component</module>
+            </modules>
+        </profile>
+    </profiles>
+
+    <properties>
+        <siddhi.version>4.0.0-M89</siddhi.version>
```
Comment (Anoukh) [inline pom.xml]: Bump to M107

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        <commons-lang3.version>3.3.2</commons-lang3.version>
+        <log4j.version>1.2.17.wso2v1</log4j.version>
+        <testng.version>6.8</testng.version>
+        <jacoco.version>0.7.8</jacoco.version>
```
Comment (Anoukh) [inline pom.xml]: Bump to 0.7.9

File: component/pom.xml [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+ ~ limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.wso2.extension.siddhi.execution.graph</groupId>
+        <artifactId>siddhi-execution-graph-parent</artifactId>
+        <version>4.0.4-SNAPSHOT</version>
```
Comment (Anoukh) [inline component/pom.xml]: We should start from 1.0.0-SNAPSHOT?

File: component/pom.xml [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+    </dependencies>
+
```
Comment (Anoukh) [inline component/pom.xml]: Add the documentation-deploy profile to deploy mkdocs github page

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+    <parent>
+        <groupId>org.wso2</groupId>
+        <artifactId>wso2</artifactId>
+        <version>5</version>
+    </parent>
+
+    <groupId>org.wso2.extension.siddhi.execution.graph</groupId>
+    <artifactId>siddhi-execution-graph-parent</artifactId>
+    <version>4.0.4-SNAPSHOT</version>
```
Comment (Anoukh) [inline pom.xml]: Version should 1.0.0-SNAPSHOT

**Review by suhothayan — APPROVED** [review]

**Pull request merged**